controllers/user/index.js

```diff
@@ -0,0 +1,109 @@
+                    return res.json(data)
+                })
+                .on(ERROR, next)
+                .on(VALIDATION_ERROR, next)
+                .on(NOT_FOUND, next)
+        } catch (error) {
+            //handle error
+        }
+        userOperation.login(req.body)
+    },
+
+    async getUser(req, res, next) {
+        const userOperation = new UserOperation();
+        const {
+            SUCCESS,
+            ERROR,
+            VALIDATION_ERROR,
+            NOT_FOUND
+        } = userOperation.outputs
+        try {
+            userOperation.on(SUCCESS, (data) => {
+                    return res.json(data)
+                })
+                .on(ERROR, next)
+                .on(VALIDATION_ERROR, next)
+                .on(NOT_FOUND, next)
+        } catch (error) {
+            //handle error
+        }
+        userOperation.getUser(req.body,req.headers.authorization)
+    },
+
+    async addUser(req, res, next) {
+        const userOperation = new UserOperation();
+        const {
+            SUCCESS,
+            ERROR,
+            VALIDATION_ERROR,
+            NOT_FOUND
+        } = userOperation.outputs
+        try {
+            userOperation.on(SUCCESS, (data) => {
+                    return res.json(data)
+                })
+                .on(ERROR, next)
+                .on(VALIDATION_ERROR, next)
+                .on(NOT_FOUND, next)
+        } catch (error) {
+            //handle error 
+        }
+        if(req.file){
+            userOperation.addUser(req.body,req.headers.authorization,req.file.filename )
+        }
+        else
+        userOperation.addUser(req.body,req.headers.authorization,'avatar.png' )
+    },
+    async updateProfile(req, res, next) {
+        const userOperation = new UserOperation();
+        const {
+            SUCCESS,
+            ERROR,
+            VALIDATION_ERROR,
+            NOT_FOUND
+        } = userOperation.outputs
+        try {
+            userOperation.on(SUCCESS, (data) => {
+                    return res.json(data)
+                })
+                .on(ERROR, next)
+                .on(VALIDATION_ERROR, next)
+                .on(NOT_FOUND, next)
+        } catch (error) {
+            //handle error
+        }
+        userOperation.updateProfile(req.body,req.headers.authorization)
+    },
+    async updateUser(req, res, next) {
+        const userOperation = new UserOperation();
+        const {
+            SUCCESS,
+            ERROR,
+            VALIDATION_ERROR,
+            NOT_FOUND
+        } = userOperation.outputs
+        try {
+            userOperation.on(SUCCESS, (data) => {
+                    return res.json(data)
+                })
+                .on(ERROR, next)
+                .on(VALIDATION_ERROR, next)
+                .on(NOT_FOUND, next)
+        } catch (error) {
+            //handle error
+        }
+        userOperation.updateUser(req.body,req.headers.authorization)
```

Comment:
add await here in front of the function call
Authorization should be handled in the controller